core/src/components/alert/alert.ios.scss

```diff
@@ -286,13 +286,13 @@
   overflow: hidden;
 }
 
-.alert-button:first-child {
+:host .alert-button:first-child {
```

Comment:
Why was this added?

Reply:
Without `host-context` then it leads to compiling the wrong css. Notice the lack of space between `[dir=rtl]` and `.sc-ion-alert-ios`

```css
[dir=rtl].sc-ion-alert-ios .alert-button.sc-ion-alert-ios:last-child {
 border-right:0.55px solid rgba(var(--ion-text-color-rgb, 0, 0, 0), 0.2)
}
```

By adding `:host`, the compiled css has the space:

```css
[dir=rtl] .sc-ion-alert-ios-h .alert-button.sc-ion-alert-ios:last-child {
 border-right:0.55px solid rgba(var(--ion-text-color-rgb, 0, 0, 0), 0.2)
}
```

This only happens when the component is [scoped](https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/alert/alert.tsx#L43). 

If the component is not scoped or a shadow then it would have worked without `:host`:

```css
.alert-button:first-child {
  @include rtl() {
    ...
  }
}
```